test-network-function/lifecycle/suite.go

```diff
@@ -424,38 +448,45 @@ func testGracePeriod(env *config.TestEnvironment) {
 	})
 }
 
-func testShutdown(env *config.TestEnvironment) {
+func test(env *config.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestShudtownIdentifier)
+	testfunction(env, testID, relativeShutdownTestDirectoryPath, relativeShutdownTestPath, "pre-stop")
+
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestLivenessIdentifier)
+	testfunction(env, testID, relativeLivenessTestDirectoryPath, relativeLivenessTestPath, "liveness")
+
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestReadinessIdentifier)
+	testfunction(env, testID, relativeReadinessTestDirectoryPath, relativeReadinessTestPath, "readiness")
+}
+func testfunction(env *config.TestEnvironment, testID, relativeNameTestDirectoryPath, relativeNameTestPath, name string) {
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		failedPods := []*configsections.Pod{}
-		ginkgo.By("Testing PUTs are configured with pre-stop lifecycle")
+		ginkgo.By(fmt.Sprintf("Testing PUTs are configured with %s lifecycle", name))
 		for _, podUnderTest := range env.PodsUnderTest {
-			ginkgo.By(fmt.Sprintf("should have pre-stop configured %s/%s", podUnderTest.Namespace, podUnderTest.Name))
-			passed := shutdownTest(podUnderTest.Namespace, podUnderTest.Name, env.GetLocalShellContext())
+			ginkgo.By(fmt.Sprintf("should have %s configured %s/%s", name, podUnderTest.Namespace, podUnderTest.Name))
+			passed := functionTest(podUnderTest.Namespace, podUnderTest.Name, env.GetLocalShellContext(), relativeNameTestDirectoryPath, relativeNameTestPath, name)
 			if !passed {
 				failedPods = append(failedPods, podUnderTest)
 			}
 		}
 		if n := len(failedPods); n > 0 {
-			log.Debugf("Pods without pre-stop configured: %+v", failedPods)
-			ginkgo.Fail(fmt.Sprintf("%d pods do not have pre-stop configured.", n))
+			log.Debugf("Pods without %s configured: %+v", name, failedPods)
 		}
 	})
 }
-
-func shutdownTest(podNamespace, podName string, context *interactive.Context) bool {
+func functionTest(podNamespace, podName string, context *interactive.Context, relativeNameTestDirectoryPath, relativeNameTestPath, name string) bool {
```

Comment:
You have a  "test", a "testFunction" and now a "functionTest", which looks really confusing. You should use more meaningful names. These are my suggestions:
functionTest() -> gotemplateTest(): uses the paths to the gotemplates.
testFunction() -> testPods(): iterates through the pods, calling gotemplateTest() using the corresponding go templates.

As I said in a comment above, change the test() function for two individual functions: testLiveness() and testReadiness(), which should call testPods() withe corresponding testID, schemapath and the name to use in log traces.

Reply:
I saw that all the functions write like this , for example:"shutdown ", we have shutdownTest and TestShutdown, this is the reason what I wrote like this. 

and I agree to change functionTest()->gotemplateTest(),testFunction() -> testPods().
and about function test(), I wrote before that I cant to split 2 function because I getting error in make lint that have "duplicate lines", what do you think? @greyerof